src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "pb.h"
+
+typedef struct ProtoMulticastCommunicationProfile ProtoMulticastCommunicationProfile_t;
+
+/**
+ * Event Flags: used to signal tasks to unblock and run specific actions
+ */
+typedef enum
+{
+    PROTO_UPDATED   = 1 << 0,  // protobuf struct has been updated
```

Comment:
Are these special enum values required for LWIP to work? Do we need to do this bit-shifting?

Reply:
these are arbitrary event flags that I have selected to notify tasks to unblock and run

there is a comment ontop that indicates that they are flags, let me know if I should clarify it a bit more

Reply:
Thoughts on doing it like this:
```
typedef struct
{
    int takes_damage : 1;
    int grabbable    : 1;
    int liquid       : 1;
    int some_other   : 1;
} flags;
```
source: https://stackoverflow.com/questions/1631266/flags-enum-c

Reply:
this may not work because we need bit masks and not flags themselves, we could do these and then shift + OR the values later on but I don't see any drawback in creating the masks in the first place

Reply:
Ah I think I see.

So the lwip / osEvent interface allows the user to define their own event types / flags, but requires that these flags be masks / have some way of being combined and read from a single `uint32_t` (because that's the interface).

In that case I'm fine with this just would like a comment explaining that.

Reply:
yeah this is a feature of RTOS, not sure what comment I can add though, 
`Event Flags: masks used to signal tasks to unblock and run specific actions` is what I have rn

Reply:
I think the struct can be cast to a uin32_t so the specific bits can be set using the struct

Reply:
so it would be 
`flags = {1,0,0,1};`
`flags_mask = (int32_t)flags`

instead of 

`flag_mask = RECV_PROTO | SOMEOTHER_FLAG;`

for the arguable benefit of doing

`if (flags.proto_received) `
instead of 
`if(flags == PROTO_RECEIVED)`

if im undestanding this correctly?

Reply:
uint32_t flag = (uint32_t) EventFlag{.recv_proto=1, .someotherflag=1} I think

Reply:
Wouldn't that be casting the address of the struct, not the contents? I think I prefer the original enum flags because most people will be more familiar with that, and it's just as readable.

Reply:
I like the enum flags. There's less room to misunderstand IMO